### PR TITLE
HGCal - Minimum Cluster size for Pattern Recognition

### PR DESCRIPTION
**Pull request by felicepantaleo**
#### PR description:

This PR is introducing a new filter on the minimum layer clusters size that a hgcalLayerCluster requires in order to be passed to the next step of the Pattern Recognition in TICL. This is useful to make PR more robust against single cell inefficiency and to remove duplicate tracksters.
#### PR validation:

This PR was tested with 20490.0_SingleGammaPt25Eta1p7_2p7 checking that the number of hgcalLayerClusters fed to the pattern recognition is halved, and that the number of duplicate tracksters is decreased. 
Another PR will be made in order to make the Pattern Recognition by CA more robust as well.

@rovere @cseez @amartelli fyi

### CODE REVIEW

**Comment by slava77** [comment]
please test workflow 20493.52

**Comment by kpedro88** [comment]
+upgrade

File: RecoHGCal/TICL/python/ticl_iterations.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def TICL_iterations_withReco(process):
   )
 
   process.FilteredLayerClusters = filteredLayerClustersProducer.clone(
+      clusterFilter = "ClusterFilterByAlgoAndSize",
```
Comment (slava77) [inline RecoHGCal/TICL/python/ticl_iterations.py]: this is already a part of filteredLayerClustersProducer PSet. Is this redundant change really necessary here?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/27666	b3cbe30
- code changes are in line with the PR description (although it could be more clear that this is replacing an old filter, not just "introducing a new filter")
- jenkins tests pass and comparisons with the baseline show no differences because the TICL workflow is not used in comparisons in the tests

https://github.com/cms-sw/cmssw/pull/27666#pullrequestreview-272133903 is minor enough and applies only to an experimental config/customization.

**Comment by fabiocos** [comment]
+1

**Pull request merged**